### PR TITLE
Support repository urls from different SCMs in Dashboard

### PR DESCRIPTION
**Pull request by bartsmykla**
Closes https://github.com/openfaas/openfaas-cloud/issues/304

Signed-off-by: Bart Smykla <bsmykla@vmware.com>

## Description
Added support to other SCM's as repository urls inside Dashboard

## How Has This Been Tested?

Deployed to my cluster and tested with functions on GitHub and my own GitLab instance.

## How are existing users impacted? What migration steps/scripts do we need?
They are not

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

**Comment by alexellis** [comment]
This annotation won't exist for existing deployments, am I correct in thinking that? You listed the impact as none but I'm not convinced.

File: dashboard/client/src/api/functionsApi.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import axios from 'axios';
 import moment from 'moment';
 
+const getRepoURL = annotations => annotations['com.openfaas.cloud.git-repo-url'] || '';
```
Comment (alexellis) [inline dashboard/client/src/api/functionsApi.js]: From review call - could we return the hard-coded default from this method? Alternatively write a migration script and keep the code simple.

**Comment by alexellis** [comment]
https://gist.github.com/martindekov/f1656d6da27290dfeb8e7379c2e8b1a5

**Comment by alexellis** [comment]
This migration script will be needed ASAP @bartsmykla - remember to populate an annotation in your kubectl script. Once it's ready I will run it on the community cluster before deploying the new version of the dashboard.

File: dashboard/client/src/api/functionsApi.js (hunk 1 above)
Comment (bartsmykla) [reply dashboard/client/src/api/functionsApi.js]: I have sent you migration script and kept the code suimple

**Comment by bartsmykla** [comment]
@alexellis PR should be ready to merge now

**Comment by alexellis** [comment]
@bartsmykla we need to update `dashboard/README.md` which gives a sample of how to deploy functions for testing. Please add a commit with the new annotation set via the CLI command.

**Comment by alexellis** [comment]
i.e. here https://github.com/openfaas/openfaas-cloud/tree/master/dashboard#deploy-a-few-functions-for-sample-data

**Comment by bartsmykla** [comment]
@alexellis it should be fine now

**Comment by alexellis** [comment]
FYI commit messages should be wrapped to 72 chars. I'll merge this one since you've been backwards and forwards a little bit, but please try to keep to this in future.

**Comment by bartsmykla** [comment]
Forgot about that

**Comment by alexellis** [comment]
Fixing the commit message is appreciated.

**Pull request merged**